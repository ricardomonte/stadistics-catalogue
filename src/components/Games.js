import React from 'react';
import PropTypes from 'prop-types';

const Games = ({ game }) => (
  <div>
    <img alt="something" />
    <div>
      <h4>{game.title}</h4>
      <p>some text</p>
      <p />
    </div>
  </div>
);

Games.protoTypes = {
  game: PropTypes.instanceOf(Object).isRequired,
};

export default Games;
