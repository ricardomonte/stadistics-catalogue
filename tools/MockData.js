const MockData = [
  {
    id: 1,
    name: 'test data 1',
    cover: '//mock.data.com/test.jpg',
    genre: 'Strategy',
    rating: 72,
  },
  {
    id: 2,
    name: 'test data 2',
    cover: '//mock.data.com/test2.jpg',
    genre: 'Sports',
    rating: 80,
  },
  {
    id: 3,
    name: 'test data 3',
    cover: '//mock.data.com/test3.jpg',
    genre: 'Sports',
    rating: 10,
  },
  {
    id: 4,
    name: 'test data 4',
    cover: '//mock.data.com/test4.jpg',
    genre: 'Fighting',
    rating: 95.67,
  },
  {
    id: 5,
    name: 'test data 5',
    cover: '//mock.data.com/test5.jpg',
    genre: 'Arcade',
    rating: 100,
  },
];

export default MockData;
