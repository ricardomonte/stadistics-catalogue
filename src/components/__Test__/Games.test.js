import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import MockData from '../../../tools/MockData';
import Games from '../Games';

test('Content from Games component matches snapshot', () => {
  const content = render(
    <BrowserRouter>
      <Games game={MockData[0]} />
    </BrowserRouter>,
  );
  expect(content.container).toMatchSnapshot();
});
