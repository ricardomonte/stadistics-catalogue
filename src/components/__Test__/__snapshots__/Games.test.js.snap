// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content from Games component matches snapshot 1`] = `
<div>
  <div
    class="game"
  >
    <a
      class="gameLink"
      href="/game/1"
    >
      <img
        alt="something"
        class="gameImage"
        src="https://mock.data.com/test.jpg"
      />
      <div
        class="gameContainer"
      >
        <h3
          class="gameTitle"
        >
          test data 1
        </h3>
      </div>
    </a>
  </div>
</div>
`;
